shardingsphere-kernel/shardingsphere-transaction/shardingsphere-transaction-api/src/main/java/org/apache/shardingsphere/transaction/config/TransactionRuleConfiguration.java

```diff
@@ -36,4 +37,18 @@
     private final String providerType;
     
     private final Properties props;
+    
+    /**
+     * Compare to another transaction rule configuration.
+     *
+     * @param transactionRuleConfiguration transaction rule configuration
+     * @return return true if the two transactionRuleConfiguration are the same
+     */
+    public boolean compare(final TransactionRuleConfiguration transactionRuleConfiguration) {
+        if (null == transactionRuleConfiguration) {
+            return false;
+        }
+        return Objects.equals(defaultType, transactionRuleConfiguration.defaultType) && Objects.equals(providerType, transactionRuleConfiguration.providerType)
+                && Objects.equals(props, transactionRuleConfiguration.props);
```

Comment:
the props are synchronized and may caused by performance issue.
Is the method `compare` call frequently?

Reply:
It is just called when alter global rule.